script.js

```diff
@@ -52,4 +52,112 @@ function valutaConverter(event) {
   }
 }
 
-// updateing existing currency conversion rates with new rates form----------------
+// updateing existing currency conversion rates with new rates form
+// **********************************************
+// Week2 / An aaray of the currency rate objects
+// **********************************************
+
+const brandObjectArray = [
+  {
+    timestamp: Date.now(),
+    base: "EUR",
+    date: new Date().toLocaleDateString(),
+    rates: {},
+  },
+  {
+    timestamp: Date.now(),
+    base: "USD",
+    date: new Date().toLocaleDateString(),
+    rates: {},
+  },
+  {
+    timestamp: Date.now(),
+    base: "DKK",
+    date: new Date().toLocaleDateString(),
+    rates: {},
+  },
+];
+
+document
+  .getElementById("brand-currency-form-beta")
+  .addEventListener("submit", brandCurrencyBeta);
+
+function brandCurrencyBeta(event) {
+  event.preventDefault();
+
+  //getting and filling objects of array------------
+  const baseCurrency = document.getElementById("base-currency-beta").value;
+  const quoteName = document.getElementById("quote-name-beta").value;
+  const quoteRate = document.getElementById("quote-rate-beta").value;
+
+  if (baseCurrency === "EUR") {
```

Comment:
There is an easier way how to get the selected Base Currency via `find` method

```
  const selectedCurrencyRate = brandObjectArray.find(
    (currency) => currency.base === baseCurrency,
  )
  
   selectedCurrencyRate.rates[quoteName] = +quoteRate
```

you can delete the reset

Reply:
Thank you. Have replaced my codes with your code:
my codes:
if (baseCurrency === "EUR") {
    brandObjectArray[0].rates[quoteName] = +quoteRate;
  } else if (baseCurrency === "USD") {
    brandObjectArray[1].rates[quoteName] = +quoteRate;
  } else if (baseCurrency === "DKK") {
    brandObjectArray[2].rates[quoteName] = +quoteRate;

Reply:
I just changed the const variable name to selectedCurrencyBase:

const selectedCurrencyBase = brandObjectArray.find(
    (currency) => currency.base === baseCurrency
  );